store/src/main/java/com/nytimes/android/external/store3/util/NoopPersister.java

```diff
@@ -7,45 +7,46 @@
 import com.nytimes.android.external.store3.base.impl.MemoryPolicy;
 
 import java.util.concurrent.TimeUnit;
-
 import javax.annotation.Nonnull;
 
 import io.reactivex.Maybe;
 import io.reactivex.Single;
 
-
 /**
  * Pass-through diskdao for stores that don't want to use persister
  */
 public class NoopPersister<Raw, Key> implements Persister<Raw, Key>, Clearable<Key> {
     protected final Cache<Key, Maybe<Raw>> networkResponses;
 
     NoopPersister(MemoryPolicy memoryPolicy) {
-        this.networkResponses = CacheBuilder
-            .newBuilder()
-            .expireAfterWrite(memoryPolicy.getExpireAfterWrite(), memoryPolicy.getExpireAfterTimeUnit())
-            .build();
+        if (memoryPolicy.hasAccessPolicy()) {
+            networkResponses = CacheBuilder.newBuilder()
+                .expireAfterAccess(memoryPolicy.getExpireAfterAccess(), memoryPolicy.getExpireAfterTimeUnit())
+                .build();
+
+        } else if (memoryPolicy.hasWritePolicy()) {
+            networkResponses = CacheBuilder.newBuilder()
+                .expireAfterWrite(memoryPolicy.getExpireAfterWrite(), memoryPolicy.getExpireAfterTimeUnit())
+                .build();
+        } else {
+            throw new IllegalArgumentException("No expiry policy set on memory-policy.");
+        }
```

Comment:
Technically `CacheBuilder` lets you set both `expireAfterAccess` and `expireAfterWrite` although it gives preference to `expireAfterAccess`. It's a bit of an edge case, but maybe update this block to handle both being set.

Reply:
Thanks for the review! If both are set, should we throw an exception?